Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+@Library('ystv-jenkins')
+
+String registryEndpoint = 'registry.comp.ystv.co.uk'
+
+def image
+String imageName = "ystv/showtime:${env.BRANCH_NAME}-${env.BUILD_ID}"
+
+pipeline {
+  agent {
+    label 'docker'
+  }
+
+  environment {
+    DOCKER_BUILDKIT = '1'
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        ciSkip(action: 'check')
+      }
+    }
+
+    stage('Build image') {
+      steps {
+        script {
+          docker.withRegistry('https://' + registryEndpoint, 'docker-registry') {
+            image = docker.build(imageName)
```

Comment:
Oooh, I didn't know you could do that!

Reply:
Yeah it is quite nifty that! It feels a lot nicer working with objects. Although from reading stackoverflow, not everyone is fan of it, but decided to keep it like that since it does look a bit cleaner